modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/scanner/InboxScannerService.java

```diff
@@ -305,11 +330,28 @@ public static String getCfg(Dictionary d, String key) throws ConfigurationExcept
    * @throws ConfigurationException
    *           key does not exist or is not an integer
    */
-  public static int getCfgAsInt(Dictionary d, String key) throws ConfigurationException {
+  public static Integer getCfgAsInt(Dictionary d, String key) throws ConfigurationException {
     try {
       return Integer.parseInt(getCfg(d, key));
     } catch (NumberFormatException e) {
       throw new ConfigurationException(key, "not an integer");
     }
   }
+
+  public static int getCfgAsIntOrDefault(Dictionary d, String key, int defaultValue) throws ConfigurationException {
+    String valStr = getCfgOrDefault(d, key, null);
+    if (StringUtils.isEmpty(valStr))
+      return defaultValue;
+    try {
+      return Integer.parseInt(valStr);
+    } catch (Exception e) {
+      logger.warn("The inbox configuration {} value {} is not an integer. Use default value of {}", key, valStr,
+              defaultValue);
```

Comment:
I'm not a huge fan of running with actual errors and would recommend to rather fail fast.
While it might seem nice that things still work even though the configuration is broken, it can lead to unexpected, weird behavior.

Hence, my suggestion here would be to just go with something like:

```java
if (StringUtils.isEmpty(valStr)) {
  return defaultValue;
} else {
  return Integer.parseInt(valStr);
}
```

Reply:
I implemented this change